dataset/histogram.cpp

```diff
@@ -127,6 +127,15 @@ bool is_histogram(const DataArrayConstView &a, const Dim dim) {
          coords[dim].dims()[dim] == dims[dim] + 1;
 }
 
+/// Return true if the dataset represents a histogram for given dim.
+bool is_histogram(const DatasetConstView &a, const Dim dim) {
+  const auto dims = a.dims();
+  const auto coords = a.coords();
+  return dims.find(dim) != dims.end() && coords.contains(dim) &&
+         coords[dim].dims().contains(dim) &&
+         coords[dim].dims()[dim] == dims.at(dim) + 1;
+}
+
```

Comment:
Could avoid code duplication by using a templated `is_histogram_impl`. If you replace `dims.contains(dim)` by `dims.count(dim)` that should also work with the `std::unordered_map` we are getting for dataset dims.